net/redisclient.go

```diff
@@ -221,7 +221,8 @@ func NewRedisRingClient(ro *RedisOptions) *RedisRingClient {
 		}
 
 		if ro.AddrUpdater != nil {
-			ringOptions.Addrs = createAddressMap(ro.AddrUpdater())
+			address, _ := ro.AddrUpdater()
```

Comment:
Check error and retry with a loop 5 times with backoff of some seconds and if error persist fatal at startup.

Reply:
But do we want to fail startup in case of unavailable redis?

Reply:
@AlexanderYastrebov yes we want to fail because it can show a split brain network, which we do not want